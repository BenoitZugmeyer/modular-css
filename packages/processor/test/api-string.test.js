"use strict";

const namer = require("@modular-css/test-utils/namer.js");

const Processor = require("../processor.js");

describe("/processor.js", () => {
    describe(".string()", () => {
        let processor;

        beforeEach(() => {
            processor = new Processor({
                namer,
            });
        });

        it("should process a string", async () => {
            const result = await processor.string(
                "./simple.css", ".wooga { }"
            );

            expect(result.exports).toMatchSnapshot();
            expect(result.details.exports).toMatchSnapshot();
            expect(result.details.text).toMatchSnapshot();
            expect(result.details.processed.root.toResult().css).toMatchSnapshot();
        });
    });
});
