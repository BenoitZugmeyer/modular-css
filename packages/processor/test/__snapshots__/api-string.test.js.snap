// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/processor.js .string() should process a string 1`] = `
Object {
  "wooga": Array [
    "wooga",
  ],
}
`;

exports[`/processor.js .string() should process a string 2`] = `
Object {
  "wooga": Array [
    "wooga",
  ],
}
`;

exports[`/processor.js .string() should process a string 3`] = `".wooga { }"`;

exports[`/processor.js .string() should process a string 4`] = `".wooga { }"`;
